mailpoet/lib/Config/Populator.php

```diff
@@ -238,8 +238,12 @@ private function createDefaultSettings() {
     }
 
     // set default sender info based on current user
+    $currentUserName = $currentUser->display_name ?: ''; // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+    // parse current user name if an email is used
+    $senderName = explode('@', $currentUserName);
+    $senderName = reset($senderName);
```

Comment:
I'd say we can do even more like checking for other [allowed characters](https://en.wikipedia.org/wiki/Email_address#Local-part) in email addresses and trying to use space (?) instead of them, what do you think?

Reply:
I think this is unnecessary because parsing can happen only when the WP user that activates the plugin for the first time has an email address in the name. Concurrently the user can modify the name in the form input without any validation.